File: 10866.cpp
#pragma warning(disable:4996)
#include <iostream>
#include <queue>
#include <string>
using namespace std;

int main() {

	int N;
	cin >> N;

	deque<int> dq;
	string str;
	int num;
	
	for (int i = 0; i < N; i++) {
		cin >> str;

		if (str == "push_front") {
			cin >> num;
			dq.push_front(num);
		}
		if (str == "push_back") {
			cin >> num;
			dq.push_back(num);
		}
		if (str == "pop_front") {
			if (dq.empty())
				cout << "-1\n";
			else {
				cout << dq.front() << "\n";
				dq.pop_front();
			}
		}
		if (str == "pop_back") {
			if (dq.empty())
				cout << "-1\n";
			else {
				cout << dq.back() << "\n";
				dq.pop_back();
			}
		}

		if (str == "size")
			cout << dq.size() << "\n";
		if (str == "empty")
			cout << dq.empty() << "\n";
		
		if (str == "front") {
			if (dq.empty())
				cout << "-1\n";
			else
				cout << dq.front() << "\n";
		}

		if (str == "back") {
			if (dq.empty())
				cout << "-1\n";
			else
				cout << dq.back() << "\n";
		}
	}


	return 0;
}

File: 10866.py
import sys
from collections import deque

input = sys.stdin.readline
N = int(input())

nums = deque()
top = -1

for _ in range(N):
    order = input()

    if "push_front" in order:
        ord, num = order.split()
        nums.appendleft(num)
        top += 1
    elif "push_back" in order:
        ord, num = order.split()
        nums.append(num)
        top += 1
    elif "pop_front" in order:
        if top == -1:
            print("-1")
        else:
            print(nums.popleft())
            top -= 1
    elif "pop_back" in order:
        if top == -1:
            print("-1")
        else:
            print(nums.pop())
            top -= 1
    elif "size" in order:
        print(top+1)
    elif "empty" in order:
        if top == -1:
            print("1")
        else:
            print("0")
    elif "front" in order:
        if top == -1:
            print("-1")
        else:
            print(nums[0])
    elif "back" in order:
        if top == -1:
            print("-1")
        else:
            print(nums[top])
